### PR TITLE
Feature/vsp 1259

### PR DESCRIPTION
**Pull request by mcinbell**
Paysafecard gateway

### CODE REVIEW

File: src/PaysafecardGateway.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<?php
+namespace Omnipay\Noire;
```
Comment (ybenzine) [inline src/PaysafecardGateway.php]: Would it make more sense to move all the gateways to their respective folders / namespaces?

File: src/PaysafecardGateway.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    /**
+     * Returns an authorization request for paysafecard payment type
+     *
+     * @param array $data
+     * @return \Omnipay\Noire\Message\AuthorizeRequest
```
Comment (ybenzine) [inline src/PaysafecardGateway.php]: Isn't it returning a `PaysafecardAuthorizeRequest`?

File: src/PaysafecardGateway.php (hunk 1 above)
Comment (mcinbell) [inline src/PaysafecardGateway.php]: I actually experimented with this... :) However, `Omnipay::create` didn't seem to work if I did that...

File: src/Message/Paysafecard/AuthorizeRequest.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+class AuthorizeRequest extends OppwaBaseRequest
+{
+    const PAYMENT_BRAND = 'PAYSAFECARD';
+
+    /**
+     * Sets the shopper result url, which is link used on return button loaded in redirect url
+     *
+     * @param string $value
+     */
+    public function setShopperResultUrl($value)
```
Comment (mcinbell) [inline src/Message/Paysafecard/AuthorizeRequest.php]: some we could do at some point yes... though this PR is about the Gateway... :)

**Pull request merged**